### ISSUE TITLE
TBB can't be found

### ISSUE THREAD

**Issue by mboisson** [woodem/woo#33]
Hi,
It looks like the way you detect TBB is not standard. It expects files which do not exist: 
```
CMake Error at CMakeLists.txt:96 (find_package):
  By not providing "FindTBB.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "TBB", but
  CMake did not find one.

  Could not find a package configuration file provided by "TBB" with any of
  the following names:

    TBBConfig.cmake
    tbb-config.cmake

  Add the installation prefix of "TBB" to CMAKE_PREFIX_PATH or set "TBB_DIR"
  to a directory containing one of the above files.  If "TBB" provides a
  separate development package or SDK, be sure it has been installed.
``` 

but none of the installed versions of TBB which we have provide these files. The available CMake files are: 
```
$ find /cvmfs/soft.computecanada.ca/easybuild/software/*/{avx2/,}Core/tbb/* -name '*.cmake'
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/TBBBuild.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/TBBGet.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/TBBInstallConfig.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/TBBMakeConfig.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/tbb_config_generator.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/tbb/2020.2/cmake/tbb_config_installer.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/tbb/2018_U5/cmake/TBBBuild.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/tbb/2018_U5/cmake/TBBGet.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/tbb/2018_U5/cmake/TBBMakeConfig.cmake
/cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/tbb/2018_U5/cmake/tbb_config_generator.cmake
```

**Comment by eudoxos** [comment]
Hi,

there is [FindTBB.cmake in the source directory](https://github.com/woodem/woo/blob/master/FindTBB.cmake), maybe you have some setting for cmake where it looks for files and which does not include the source root dir?

Packaged versions of TBB seem to be providing this file now (which was not the case before, as I remember), e.g. [Debian buster](https://packages.debian.org/buster/amd64/libtbb-dev/filelist) but newer packages [Debian sid](https://packages.debian.org/sid/amd64/libtbb-dev/filelist) provide the more structured files you mention.

**Comment by mboisson** [comment]
We have built literally hundreds of CMake recipes with our cmake config. I tried both building inside a build directory and in the source directory. Looks like something is missing in your CMakeLists.txt to make it look in the source directory.

**Comment by eudoxos** [comment]
I understand the need for consistency and predictability. Would you share a link to a project which uses CMake and can be built on your clusters? That might be a good inspiration for something standard and predictable.

**Comment by mboisson** [comment]
I opened a pull request to fix this

**Comment by eudoxos** [comment]
Shoule be fixed in 343f3547450f3fc6e14462fce4f516ce410cfca0 . Please re-open if you are still hitting the issue.
